projects/plugins/jetpack/_inc/lib/admin-pages/class-jetpack-redux-state-helper.php

```diff
@@ -248,7 +247,7 @@ public static function get_initial_state() {
 			'isSubscriptionSiteEnabled'     => apply_filters( 'jetpack_subscription_site_enabled', false ),
 			'newsletterDateExample'         => gmdate( get_option( 'date_format' ), time() ),
 			'subscriptionSiteEditSupported' => $current_theme->is_block_theme(),
-			'isMasterbarActive'             => ( new Modules() )->is_active( 'masterbar', false ),
+			'isMasterbarActive'             => $host->is_woa_site(),
```

Comment:
The `isMasterbarActive` var was only needed to show the masterbar notice on self-hosted sites which won't be longer the case:

https://github.com/Automattic/jetpack/blob/347b86a9713eb9909c698fb36cffcb7d2ce9bf29/projects/plugins/jetpack/_inc/client/components/jetpack-notices/index.jsx#L315

Maybe you can clean up that code too?

Reply:
Sure! Done via https://github.com/Automattic/jetpack/pull/39406/commits/cfb16c77d7bb20100e1f52fbe77b49eafbb633ef

That said, in order to clean up the Masterbar notice I also had to remove the generic logic for those deprecation notices because Masterbar was the last one. Ideally this logic should be independent of the actual feature we deprecate, otherwise we'll have to keep going back and forth upon deprecating other features. Same goes for `class-deprecate.php` cc @sergeymitr @coder-karen